src/commonMain/kotlin/exchange.dydx.abacus/state/v2/supervisor/VaultSupervisor.kt

```diff
@@ -0,0 +1,189 @@
+package exchange.dydx.abacus.state.v2.supervisor
+
+import exchange.dydx.abacus.protocols.LocalTimerProtocol
+import exchange.dydx.abacus.state.model.TradingStateMachine
+import exchange.dydx.abacus.state.model.onMegaVaultPnl
+import exchange.dydx.abacus.state.model.onVaultMarketPnls
+import exchange.dydx.abacus.state.model.onVaultMarketPositions
+import exchange.dydx.abacus.utils.AnalyticsUtils
+import exchange.dydx.abacus.utils.CoroutineTimer
+import exchange.dydx.abacus.utils.NUM_PARENT_SUBACCOUNTS
+import indexer.codegen.IndexerAssetPositionResponseObject
+import indexer.codegen.IndexerMegavaultHistoricalPnlResponse
+import indexer.codegen.IndexerMegavaultPositionResponse
+import indexer.codegen.IndexerPerpetualPositionResponseObject
+import indexer.codegen.IndexerPerpetualPositionStatus
+import indexer.codegen.IndexerPnlTicksResponseObject
+import indexer.codegen.IndexerPositionSide
+import indexer.codegen.IndexerVaultHistoricalPnl
+import indexer.codegen.IndexerVaultPosition
+import indexer.codegen.IndexerVaultsHistoricalPnlResponse
+import kotlinx.datetime.Instant
+import kotlinx.serialization.encodeToString
+import kotlinx.serialization.json.Json
+
+internal class VaultSupervisor(
+    stateMachine: TradingStateMachine,
+    helper: NetworkHelper,
+    analyticsUtils: AnalyticsUtils,
+    internal val configs: VaultConfigs,
+) : NetworkSupervisor(stateMachine, helper, analyticsUtils) {
+
+    private var indexerTimer: LocalTimerProtocol? = null
+        set(value) {
+            if (field !== value) {
+                field?.cancel()
+                field = value
+            }
+        }
+    private val megaVaultPnlPollingDuration = 60.0
+
+    override fun didSetIndexerConnected(indexerConnected: Boolean) {
+        super.didSetIndexerConnected(indexerConnected)
+
+        if (!configs.retrieveVault) {
+            return
+        }
+
+        if (indexerConnected) {
+            val timer = helper.ioImplementations.timer ?: CoroutineTimer.instance
+            indexerTimer = timer.schedule(delay = 0.0, repeat = megaVaultPnlPollingDuration) {
+                if (readyToConnect) {
+                    retrieveMegaVaultPnl()
```

Comment:
If needed, we can add a state machine function to trigger an update in the future.  It's currently just timer-based. @mike-dydx